pkg/cmd/server/origin/asset.go

```diff
@@ -81,9 +81,15 @@ func (c *AssetConfig) Run() {
 
 	go util.Forever(func() {
 		if isTLS {
+			extraCerts, err := configapi.GetNamedCertificateMap(c.Options.ServingInfo.NamedCertificates)
+			if err != nil {
+				glog.Fatal(err)
+			}
 			server.TLSConfig = &tls.Config{
 				// Change default from SSLv3 to TLSv1.0 (because of POODLE vulnerability)
 				MinVersion: tls.VersionTLS10,
+				// Set additional SNI certificates, if provided
+				NameToCertificate: extraCerts,
```

Comment:
Wait, didn't the comment from `node_config.go` just tell me that this wouldn't work?